### PR TITLE
Ecto 3.0-rc.1 support

### PR DESCRIPTION
**Pull request by rustamtolipov**
Mainly changed Ecto.Date to Date and replaced Poison with Jason per [http://blog.plataformatec.com.br/2018/10/a-sneak-peek-at-ecto-3-0-breaking-changes/](http://blog.plataformatec.com.br/2018/10/a-sneak-peek-at-ecto-3-0-breaking-changes/). 
Also fixed some deprecation warnings for `Multi.run`

### CODE REVIEW

**Comment by izelnakri** [comment]
awesome, thanks for your work! I'll review this soon.

File: example/lib/company.ex [hunk 1]
```diff
@@ -15,10 +15,10 @@ defmodule Company do
 
     has_many :people, Person
 
-    timestamps
+    timestamps()
   end
 
-  @optional_fields ~w(name is_active website city address facebook twitter founded_in)
+  @optional_fields ~w(name is_active website city address facebook twitter founded_in)a
```
Comment (izelnakri) [inline example/lib/company.ex]: what does this a do here?

File: example/.iex.exs [hunk 2]
```diff
@@ -1 +0,0 @@
-import Ecto.Query
```
Comment (izelnakri) [inline example/.iex.exs]: this is to import Ecto.Query functions globally(to Kernel) I dont think its redundant, so I would like to keep it.

File: example/mix.exs [hunk 3]
```diff
@@ -31,10 +31,11 @@ defmodule Example.Mixfile do
...
     [
-      {:postgrex, ">= 0.0.0"},
-      {:ecto, ">= 2.0.2"},
-      {:poison, ">= 2.1.0"},
-      {:paper_trail, ">= 0.7.3"}
+      {:ecto, "~> 3.0-rc", override: true},
+      {:ecto_sql, "~> 3.0-rc", override: true},
+      {:postgrex, ">= 0.0.0-rc"},
+      {:jason, "~> 1.0"},
+      {:paper_trail, path: "../"}
```
Comment (izelnakri) [inline example/mix.exs]: 👍 very nice

File: mix.exs [hunk 4]
```diff
@@ -26,10 +26,11 @@ defmodule PaperTrail.Mixfile do
 
   defp deps do
     [
-      {:ecto, ">= 2.1.0"},
-      {:poison, ">= 3.1.0 or >= 2.0.0"},
+      {:ecto, "~> 3.0-rc", override: true},
+      {:ecto_sql, "~> 3.0-rc", override: true},
       {:ex_doc, ">= 0.0.0", only: :dev},
-      {:postgrex, "~> 0.13.0"}
+      {:postgrex, ">= 0.0.0-rc"},
```
Comment (izelnakri) [inline mix.exs]: 0.0.0-rc ?

**Comment by izelnakri** [comment]
Great work overall, could you also update the README/documentation regarding the changes? Thank you!

File: example/lib/company.ex (hunk 1 above)
Comment (rustamtolipov) [reply example/lib/company.ex]: Ecto 3.0 Ecto.Changeset.cast/3 with list of strings is deprecated, you have to pass list of atoms  
[https://github.com/elixir-ecto/ecto/blob/master/CHANGELOG.md#deprecations](https://github.com/elixir-ecto/ecto/blob/master/CHANGELOG.md#deprecations)

File: example/.iex.exs (hunk 2 above)
Comment (rustamtolipov) [reply example/.iex.exs]: I forgot to put it back, sorry.

File: mix.exs (hunk 4 above)
Comment (rustamtolipov) [reply mix.exs]: I think I just copied it and pasted from mix.exs of generated phoenix 1.4.0-rc app. Do you want me to change it to the latest version or "=> 0.0.0"?

**Pull request merged**